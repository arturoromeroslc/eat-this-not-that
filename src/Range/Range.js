import React, { Component } from 'react'
import PropTypes from 'prop-types';
import './Range.css'

const propTypes = {
  onhandleFilterRange: PropTypes.func
}

export default class Range extends Component {
  constructor(props) {
    super(props)
    this.handleOnChange = this.handleOnChange.bind(this)
  }

  componentDidMount() {
    this.refs.minVal.value = parseInt(this.props.valueMin, 10)
    this.refs.maxVal.value = parseInt(this.props.valueMax, 10)
  }

  handleOnChange(valueKey, event) {
    let tmp, backendString, rangeFilter,
      max = parseInt(this.refs.maxVal.value, 10),
      min = parseInt(this.refs.minVal.value, 10)
      
    if ( min > max ) { 
      tmp = max; 
      max = min; 
      min = tmp; 
    }
    
    rangeFilter = {
      valueMin: min,
      valueMax: max
    }
    
    backendString = `gte ${min}, lte ${max}`
    this.props.onhandleFilterRange(backendString, 'calories', rangeFilter)
  }

	render () {
		return (
      <div className="range-slider">
        <div className="range-value-container">
          <span className="range-values">Min: {this.props.valueMin}</span>
          <span className="range-values">Max: {this.props.valueMax}</span>
        </div>
        <input ref="minVal" min="0" max="5000" step="200" type="range" onChange={this.handleOnChange}/>
        <input ref="maxVal" min="0" max="5000" step="200" type="range" onChange={this.handleOnChange}/>
      </div>
		)
	}
}

Range.propTypes = propTypes