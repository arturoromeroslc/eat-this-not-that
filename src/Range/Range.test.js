import React from 'react'
import Range from './Range'
import renderer from 'react-test-renderer'

test('Range snapshot test', () =>{
	const component = renderer.create(<Range />)
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot()
})