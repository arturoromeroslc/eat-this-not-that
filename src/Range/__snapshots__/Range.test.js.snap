exports[`test Range snapshot test 1`] = `
<div
  className="range-slider">
  <div
    className="range-value-container">
    <span
      className="range-values">
      Min: 
      0
    </span>
    <span
      className="range-values">
      Max: 
      0
    </span>
  </div>
  <input
    max="5000"
    min="0"
    onChange={[Function]}
    step="200"
    type="range" />
  <input
    max="5000"
    min="0"
    onChange={[Function]}
    step="200"
    type="range" />
</div>
`;
